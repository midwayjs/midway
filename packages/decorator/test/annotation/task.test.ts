import {
  Task,
  TaskLocal,
  getClassMetadata,
  Queue,
  FORMAT,
  MODULE_TASK_METADATA,
  MODULE_TASK_TASK_LOCAL_OPTIONS,
  MODULE_TASK_QUEUE_OPTIONS,
} from '../../src';

@Queue()
export class HelloTask {
  async execute(params){
    console.log(params);
  }
}

class CustomFramework {

  @Task({
    repeat: { cron: FORMAT.CRONTAB.EVERY_MINUTE}
  })
  runTask() {}

  @TaskLocal(FORMAT.CRONTAB.EVERY_MINUTE)
  runLocalTask() {}
}


describe('/test/annotation/task.test.ts', () => {
  it('test task decorator', () => {
    const meta = getClassMetadata(MODULE_TASK_METADATA, CustomFramework);
    expect(meta).toMatchSnapshot();

    const meta1 = getClassMetadata(MODULE_TASK_TASK_LOCAL_OPTIONS, CustomFramework);
    expect(meta1).toMatchSnapshot();

    const meta2 = getClassMetadata(MODULE_TASK_QUEUE_OPTIONS, CustomFramework);
    expect(meta2).toMatchSnapshot();
  });
});
