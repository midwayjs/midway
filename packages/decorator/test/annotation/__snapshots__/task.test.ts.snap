// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/test/annotation/task.test.ts test task decorator 1`] = `
Array [
  Object {
    "name": "CustomFramework",
    "options": Object {
      "repeat": Object {
        "cron": "0 * * * * *",
      },
    },
    "propertyKey": "runTask",
    "value": [Function],
  },
]
`;

exports[`/test/annotation/task.test.ts test task decorator 2`] = `
Array [
  Object {
    "options": "0 * * * * *",
    "propertyKey": "runLocalTask",
    "value": [Function],
  },
]
`;

exports[`/test/annotation/task.test.ts test task decorator 3`] = `undefined`;
